zebra-consensus/src/block/tests.rs

```diff
@@ -413,6 +413,89 @@ fn founders_reward_validation_failure() -> Result<(), Report> {
     Ok(())
 }
 
+#[test]
+fn funding_stream_validation() -> Result<(), Report> {
+    zebra_test::init();
+
+    funding_stream_validation_for_network(Network::Mainnet)?;
+    funding_stream_validation_for_network(Network::Testnet)?;
+
+    Ok(())
+}
+
+fn funding_stream_validation_for_network(network: Network) -> Result<(), Report> {
+    let blocks = match network {
+        Network::Mainnet => vec![
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_1046400_BYTES[..])
+                .expect("block should deserialize"),
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_1046401_BYTES[..])
+                .expect("block should deserialize"),
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_1180900_BYTES[..])
+                .expect("block should deserialize"),
+        ],
+        Network::Testnet => vec![
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_TESTNET_1116000_BYTES[..])
+                .expect("block should deserialize"),
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_TESTNET_1116001_BYTES[..])
+                .expect("block should deserialize"),
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_TESTNET_1326100_BYTES[..])
+                .expect("block should deserialize"),
+        ],
```

Comment:
Test coverage: Can we test on all the `MAINNET_BLOCKS` and `TESTNET_BLOCKS`, or do we need shielded coinbase for that?

Reply:
https://github.com/ZcashFoundation/zebra/pull/3017/commits/8ac0accde0f734420e1fb001d46fe7cdc82e9dbb